lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -0,0 +1,457 @@
+--{-# OPTIONS_GHC -fno-warn-unused-imports -fno-warn-redundant-constraints
+--   -fno-warn-unused-matches -fno-warn-unused-local-binds #-}
+
+{-
+Good to read before / additional resources:
+
+- Module's documentation in `ouroboros-network/typed-protocols/src/Network/TypedProtocols.hs`
+- Data Diffusion and Peer Networking in Shelley (see: https://raw.githubusercontent.com/wiki/input-output-hk/cardano-wallet/data_diffusion_and_peer_networking_in_shelley.pdf)
+    - In particular sections 4.1, 4.2, 4.6 and 4.8
+
+
+-}
+
+module Cardano.Wallet.Shelley.Network
+    ( runExperiment
+    , point1
+      -- * Top-Level Interface
+    , ChainParameters (..)
+
+    -- * Re-Export
+    , EpochSlots (..)
+    , ProtocolMagicId (..)
+
+      -- * Constructors
+    , connectClient
+    , dummyNodeToClientVersion
+
+      -- * Transport Helpers
+    , localSocketAddrInfo
+    , localSocketFilePath
+    ) where
+
+import Prelude
+
+import Cardano.Chain.Slotting
+    ( EpochSlots (..) )
+import Cardano.Crypto
+    ( ProtocolMagicId (..) )
+import Codec.SerialiseTerm
+    ( CodecCBORTerm )
+import Network.Mux.Interface
+    ( AppType (..) )
+import Network.Mux.Types
+    ( MuxError )
+import Network.TypedProtocol.Channel
+    ( Channel )
+import Network.TypedProtocol.Codec
+    ( Codec )
+import Network.TypedProtocol.Codec.Cbor
+    ( DeserialiseFailure )
+import Network.TypedProtocol.Driver
+    ( TraceSendRecv, runPeer )
+import Ouroboros.Consensus.Ledger.Byron
+    ( ByronBlock (..)
+    , ByronHash (..)
+    , GenTx
+    , decodeByronBlock
+    , decodeByronGenTx
+    , decodeByronHeaderHash
+    , encodeByronBlock
+    , encodeByronGenTx
+    , encodeByronHeaderHash
+    )
+import Ouroboros.Consensus.NodeId
+    ( CoreNodeId (..) )
+import Ouroboros.Network.Block
+    ( Point, Tip (..) )
+import Ouroboros.Network.Magic
+    ( NetworkMagic (..) )
+import Ouroboros.Network.Mux
+    ( OuroborosApplication (..) )
+import Ouroboros.Network.NodeToClient
+    ( ConnectionId (..)
+    , NetworkConnectTracers (..)
+    , NodeToClientProtocols (..)
+    , NodeToClientVersion (..)
+    , NodeToClientVersionData (..)
+    , connectTo
+    , localTxSubmissionClientNull
+    , nodeToClientCodecCBORTerm
+    )
+import Ouroboros.Network.Protocol.ChainSync.Client
+    ( ChainSyncClient (..)
+    , ClientStIdle (..)
+    , ClientStIntersect (..)
+    , ClientStNext (..)
+    , chainSyncClientPeer
+    )
+import Ouroboros.Network.Protocol.ChainSync.Codec
+    ( codecChainSync )
+import Ouroboros.Network.Protocol.ChainSync.Type
+    ( ChainSync )
+import Ouroboros.Network.Protocol.Handshake.Version
+    ( DictVersion (..), simpleSingletonVersions )
+import Ouroboros.Network.Protocol.LocalTxSubmission.Client
+    ( LocalTxSubmissionClient (..), localTxSubmissionClientPeer )
+import Ouroboros.Network.Protocol.LocalTxSubmission.Codec
+    ( codecLocalTxSubmission )
+import Ouroboros.Network.Protocol.LocalTxSubmission.Type
+    ( LocalTxSubmission )
+
```

Comment:
why space here ?

Reply:
Hm, right. Mistake. Although I don't think we ever want to merge this PR? (Or am I wrong?)